### ISSUE TITLE
Ability to pass extra options (via meson.build) to static linker tool

### ISSUE THREAD

**Issue by mstorsjo** [mesonbuild/meson#10074]
It would be useful to be able to pass extra options to the static linker tool. In particular, MS lib.exe can print extra warnings like this:

    warning LNK4221: This object file does not define any previously undefined public symbols, so it will not be used by any link operation that consumes this library

If this is an expected situation, it'd nice to be able to pass `-ignore:4221` to the tool to silence this warning.

It's possible to pass such parameters via the cross file, e.g. `ar = ['lib', '-ignore:4221']`, but I don't see a way for the meson.build file to add the argument in a way similar to `add_project_arguments(cc.get_supported_arguments(optional_arguments), language : 'c')`.

**Comment by eli-schwartz** [comment]
Does `add_project_link_arguments()` or the `link_args: ...` kwarg work for you?

**Comment by mstorsjo** [comment]
> Does `add_project_link_arguments()` or the `link_args: ...` kwarg work for you?

Sorry, I should have clarified that I did test that, and arguments passed there do end up passed to the dynamic linker when linking executables/DLLs, but not when creating a static library.

**Comment by dcbaker** [comment]
yeah... we probably need a `add_project_link_args(..., static : true)`
